tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,1138 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import logging
+import os
+
+import pytest
+from botocore.exceptions import WaiterError
+
+from tests.integ.timeout import timeout_and_delete_endpoint_by_name
+from sagemaker.tensorflow import TensorFlow, TensorFlowModel, TensorFlowPredictor
+from sagemaker.utils import unique_name_from_base
+from sagemaker.workflow.model_step import (
+    ModelStep,
+    _REGISTER_MODEL_NAME_BASE,
+    _CREATE_MODEL_NAME_BASE,
+)
+from tests.integ.retry import retries
+from sagemaker.drift_check_baselines import DriftCheckBaselines
+from sagemaker import (
+    PipelineModel,
+    TrainingInput,
+    Model,
+    ModelMetrics,
+    MetricsSource,
+    get_execution_role,
+    ModelPackage,
+)
+from sagemaker import FileSource, utils
+from sagemaker.pytorch import PyTorch
+from sagemaker.s3 import S3Uploader
+from sagemaker.mxnet.model import MXNetModel
+from sagemaker.workflow.condition_step import ConditionStep
+from sagemaker.workflow.parameters import ParameterInteger, ParameterString
+from sagemaker.workflow.pipeline import Pipeline
+from sagemaker.workflow.steps import TrainingStep
+from sagemaker.xgboost import XGBoost, XGBoostModel, XGBoostPredictor
+from sagemaker.workflow.conditions import (
+    ConditionGreaterThanOrEqualTo,
+)
+from tests.integ.kms_utils import get_or_create_kms_key
+from tests.integ import DATA_DIR
+
+
+_REGISTER_MODEL_TYPE = "RegisterModel"
+_CREATE_MODEL_TYPE = "Model"
+_XGBOOST_PATH = os.path.join(DATA_DIR, "xgboost_abalone")
+_XGBOOST_TEST_DATA = "6 1:3 2:0.37 3:0.29 4:0.095 5:0.249 6:0.1045 7:0.058 8:0.067"
+_TENSORFLOW_PATH = os.path.join(DATA_DIR, "tfs/tfs-test-entrypoint-and-dependencies")
+_TENSORFLOW_TEST_DATA = {"instances": [1.0, 2.0, 5.0]}
+
+
+@pytest.fixture
+def role(pipeline_session):
+    return get_execution_role(pipeline_session)
+
+
+@pytest.fixture
+def pipeline_name():
+    return utils.unique_name_from_base("my-pipeline-model-step")
+
+
+def test_conditional_pytorch_training_model_registration(
```

Comment:
test_pytorch_training_model_registration_model_creation_without_custom_inference

Reply:
Updated in the new commit